sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_MockingReadingEventsTests.cs

```diff
@@ -0,0 +1,254 @@
+                .Returns(GetMockEventsFromPartition(mockPartitionId,mockLastEnquedEventProperties))
+                .Verifiable();
+
+            var consumer = mockEventHubConsumerClient.Object;
+
+            try
+            {
+                using CancellationTokenSource cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+                string firstPartition = (await consumer.GetPartitionIdsAsync(cancellationSource.Token)).First();
+                PartitionProperties partitionProperties = await consumer.GetPartitionPropertiesAsync(firstPartition, cancellationSource.Token);
+                EventPosition startingPosition = EventPosition.FromOffset(partitionProperties.LastEnqueuedOffset);
+
+                var options = new ReadEventOptions
+                {
+                    TrackLastEnqueuedEventProperties = true
+                };
+
+                await foreach (PartitionEvent partitionEvent in consumer.ReadEventsFromPartitionAsync(
+                    firstPartition,
+                    startingPosition,
+                    options,
+                    cancellationSource.Token))
+                {
+                    LastEnqueuedEventProperties properties =
+                        partitionEvent.Partition.ReadLastEnqueuedEventProperties();
+
+                    Assert.AreEqual(mockPartitionId, partitionEvent.Partition.PartitionId);
+                    Assert.AreEqual(lastSequence, properties.SequenceNumber);
+                    Assert.AreEqual(lastOffset, properties.Offset);
+                    Assert.AreEqual(fakeDate, properties.EnqueuedTime);
+                    Assert.AreEqual(fakeDate, properties.LastReceivedTime);
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                // This is expected if the cancellation token is
+                // signaled.
+            }
+            finally
+            {
+                await consumer.CloseAsync();
+            }
+
+            mockEventHubConsumerClient.VerifyAll();
+
+            #endregion
+        }
+
+        private IEnumerable<EventData> GenerateBatchData()
+        {
+            var generatedBatchData = new List<EventData>();
+            var properties = new Dictionary<string, object> { { "id", 12 } };
+            var systemProperties = new Dictionary<string, object> { { "custom", "sys-value" } };
+
+            for (int index = 0; index < 5; ++index)
+            {
+                var eventBody = new BinaryData($"This is an example event body #{index}");
+                generatedBatchData.Add(EventHubsModelFactory.EventData(eventBody, properties, systemProperties));
+            }
+
+            return generatedBatchData;
+        }
+
+        private IAsyncEnumerable<PartitionEvent> GetMockEventsFromPartition(string partitionId,LastEnqueuedEventProperties lastEnqueuedEventProperties = default)
+        {
```

Comment:
Hi @jsquire,
I need your input here, what is the best way to create the IAsyncEnumerable response for mocking ReadEventsFromPartitionAsync.

Reply:
When I'm using this method here, I would need to make this method "async" but then I dont have an "await" statement in the method...😅
So I was wondering is there an alternative to do this
Thanks